jwst/residual_fringe/residual_fringe_step.py

```diff
@@ -104,61 +96,42 @@ def process(self, input):
-                for model in input_models:
-                    model.meta.cal_step.outlier_detection = "SKIPPED"
-                else:
-                    input_models.meta.cal_step.outlier_detection = "SKIPPED"
-                self.skip = True
-                return input_models
-
-        # loop over each model
         # 1. read in reference file for model
-        # 2. correct each model
-        # 3. append corrected data to output_models - to return from step
-        self.output_models = datamodels.ModelContainer()
-        for model in input_models:
-            # Open the residual fringe reference file
-            self.residual_fringe_filename = self.get_reference_file(model,
-                                                                    'fringefreq')
-            self.log.info('Using Residual FRINGE reference file:{}'.
-                          format(self.residual_fringe_filename))
-
-            # Open the regions reference file
-            self.regions_filename = self.get_reference_file(model, 'regions')
-            self.log.info('Using MRS regions reference file: {}'.
-                          format(self.regions_filename))
-
-            # Check for a valid reference files. If they are not found skip step
-            if self.residual_fringe_filename == 'N/A' or self.regions_filename == 'N/A':
-
-                if self.residual_fringe_filename == 'N/A':
-                    self.log.warning('No Residual FRINGE reference file found')
-                    self.log.warning('Residual Fringe step will be skipped')
-
-                if self.regions_filename == 'N/A':
-                    self.log.warning('No MRS regions reference file found')
-                    self.log.warning('Residual Fringe step will be skipped')
-
-                if self.input_container:
-                    for model in self.input_models:
-                        model.meta.cal_step.outlier_detection = "SKIPPED"
-                        self.output_models.append(model)
-                else:
-                    input.meta.cal_step.residual_fringe = "SKIPPED"
-                    self.output_models.append(input)
-                    self.skip = True
-                return self.output_models
-
-            # Do the correction
-            rfc = residual_fringe.ResidualFringeCorrection(model,
-                                                           self.residual_fringe_filename,
-                                                           self.regions_filename,
-                                                           self.ignore_regions,
-                                                           **pars)
-            result = rfc.do_correction()
-            result.meta.cal_step.residual_fringe = 'COMPLETE'
-            self.output_models.append(result)
-
-        return self.output_models
+        # 2. correct the  model
+        # 3. return from step
+
+        # Open the residual fringe reference file
```

Comment:
Minor nit, but technically neither of the reference files is being "opened" by these calls. All you're doing is finding the names of the right files to use. They get opened later on.